drivers/storage/mountmgr/device.c

```diff
@@ -2064,11 +2064,13 @@ MountMgrVolumeMountPointChanged(IN PDEVICE_EXTENSION DeviceExtension,
 
     if (FsDeviceInfo.DeviceType != FILE_DEVICE_DISK && FsDeviceInfo.DeviceType != FILE_DEVICE_VIRTUAL_DISK)
     {
+        Status = STATUS_INVALID_PARAMETER;
         goto Cleanup;
     }
 
-    if (FsDeviceInfo.Characteristics != (FILE_REMOTE_DEVICE | FILE_REMOVABLE_MEDIA))
+    if (FsDeviceInfo.Characteristics & (FILE_REMOTE_DEVICE | FILE_REMOVABLE_MEDIA))
```

Comment:
Wait: Do we want to fail for removable media, or not?

Reply:
Windows mountmgr seems to fail on removable media (instead of failing on non-removable media), so my understanding is that we should do likewise.

Reply:
Ah yes I understand now: This `MountMgrVolumeMountPointChanged` function, helper for the `MountMgrVolumeMountPointCreated/Deleted` functions, is used as part of the management of the mountmgr mount-point database in the `System Volume Information` root directory of the drive. And indeed we want them only in fixed disks. So:
- if device is not a disk (or virtual disk), we fail;
- if device is remote (network) or removable, we fail as well.

Since the mountmgr code is overall not very much commented (like, what it's doing what it does), I'm adding code-comment suggestions in your PR. If you think they can be useful, you can add them.